### PR TITLE
add fields to Country and District

### PR DESCRIPTION
**Pull request by batpad**
This adds the `record_type` and `is_enclave` fields as per https://github.com/IFRCGo/go-api/issues/437#issuecomment-507611037

This also make the `record_type` field for Countries editable in the List View in the Admin, to make it easier for users to edit.

@szabozoltan69 let me know if you are able to test this and if this looks good. Please wait a bit for us both to test before merging - I just want to make sure adding these fields does not break anything else (I don't think it should, but will be good to be sure!)

@geohacker - can I also ask you for a quick review?

### CODE REVIEW

File: api/models.py [hunk 1]
```diff
@@ -71,10 +71,21 @@ def region_name(self):
...
     return 'logos/%s/%s' % (instance.iso, filename)
 
+class CountryType(IntEnum):
+    '''
+        We use the Country model for some things that are not "Countries".
+        For eg. IFRC (Not a country) or groups of countries like Southeast Asia (A region, not a country)
+    '''
+    NOT_COUNTRY = 0
+    COUNTRY = 1
+    GROUPING = 2
```
Comment (geohacker) [inline api/models.py]: Is `grouping` the best way? Or should we store actual admin level type?

Comment (batpad) [reply api/models.py]: Yea, was not sure what to call this case -- so, we have entries that are Countries - eg. India. This is straightforward as type `1` above.

Groupings are things like:
 - `Andean Country Cluster`
 - `Asia Pacific Region`
 - `Central Asia Country Cluster`
 - `English Speaking Caribbean Country Cluster`

I'm not sure these fit neatly into admin level definitions either. Perhaps we can go with `Cluster` to be closer to how a lot of these are called in the backend?

@ElsaRaunioIFRC - do you have an idea for what "type" we should assign to things like the examples above?

**Comment by ElsaRaunioIFRC** [comment]
Hi Sanjay,

How about this type of approach?

“ENTITY”, “TYPE”
‘Afghanistan’, ‘Country’
‘Andean Country Cluster’, ‘Cluster’
‘Asia-Pacific Region’, ‘Region’
‘Nepal Country Office’, ‘Country_Office’
‘Red Cross EU Office’, ‘Representative_Office’

With this approach, we would have five types: countries, clusters, regions, country offices and representative offices. The country offices and representative offices are important mainly for the surge deployments as IFRC personnel from these offices are sometimes deployed to other operations.

Let me know if I should clarify anything

Best regards,

Elsa

**Comment by batpad** [comment]
Thank you @ElsaRaunioIFRC - that is perfect. I have made the required changes to the code. We shall let you know when this is available to see and edit in the Admin section.

**Comment by batpad** [comment]
@szabozoltan69 I added the ability to filter districts by country into this PR, ref #458

**Pull request merged**